sdk/appcenter/src/main/java/com/microsoft/appcenter/AppCenter.java

```diff
@@ -418,6 +448,36 @@ private synchronized boolean instanceConfigure(Application application, String a
         mApplication = application;
         mAppSecret = appSecret;
 
+        /* Init parsing, the app secret string can contain other secrets.  */
+        TextUtils.SimpleStringSplitter pairSplitter = new TextUtils.SimpleStringSplitter(PAIR_DELIMITER);
+        TextUtils.SimpleStringSplitter keyValueSplitter = new TextUtils.SimpleStringSplitter(KEY_VALUE_DELIMITER);
+        pairSplitter.setString(appSecret);
+
+        /* Split by pairs. */
+        for (String keyValue : pairSplitter) {
+
+            /* Split key and value. */
+            keyValueSplitter.setString(keyValue);
+            String key = keyValueSplitter.next();
+
+            /* A value with no key is default to the app secret. */
+            if (!keyValueSplitter.hasNext()){
+                if (keyValue.indexOf(KEY_VALUE_DELIMITER) == -1){
```

Comment:
same comment about reformatting code (and code analysis, imports, cleanup using android UI commit dialog)